integr8ly/keycloak-operator
# [WIP] [DO NOT MERGE] Plugins

## Motivation

Allow to specify a list of plugins in the CR that the operator will install along with Keycloak.

Would like to get some feedback on this approach.

## What

We rather not modify the template so this PR implements functions to inject the required kubernetes resources to the template after it's loaded.

This works by using an init container. The code that runs in that container can be found here: https://github.com/integr8ly/sso_plugins_init

## Why

To allow installing plugins.



## Verification Steps

Add the steps required to check this change. Following an example.

1. Deploy Keycloak and a Realm and make sure that the CR has the `plugins` field has `keycloak-metrics-spi`
2. Navigate to the `<base url>/auth/realms/master/metrics`
4. Metrics should be returned from that endpoint

## Checklist:


## Progress

- [x] Finished POC
- [ ] TODO: get feedback
